test/framework/src/main/java/org/elasticsearch/test/ESIntegTestCase.java

```diff
@@ -940,6 +942,41 @@ public void waitNoPendingTasksOnAll() throws Exception {
         assertNoTimeout(clusterAdmin().prepareHealth(TEST_REQUEST_TIMEOUT).setWaitForEvents(Priority.LANGUID).get());
     }
 
+    /**
+     * Waits for the node {@code viaNode} to see {@code masterNodeName} as the master node in the cluster state and asserts that.
+     * Note that this does not guarantee that all other nodes in the cluster are on the same cluster state version already.
+     *
+     * @param viaNode the node to check the cluster state one
+     * @param masterNodeName the master node name that we wait for
+     */
+    public void awaitAndAssertMasterNode(String viaNode, String masterNodeName) throws Exception {
```

Comment:
I added these methods in `ESIntegTestCase` because I see other use cases for these methods (which I'll get to in follow-up PRs). I'm a little bit on the fence about which class they should live in, though. I would actually like to avoid adding them in this class, as this class is too big already. Maybe `InternalTestCluster`? Or an entirely new class?